app/knight_fighting/battle.py

```diff
@@ -0,0 +1,26 @@
+from .knight import Knight
+
+
+class Battle:
+    def __init__(self, first: Knight, second: Knight) -> None:
+        self.first = first
+        self.second = second
+        self.winner = None
+
+    def simulate_battle(self) -> None:
+        self.first.hp -= self.second.power - self.first.protection
+        self.second.hp -= self.first.power - self.second.protection
+
+        if self.first.hp <= 0:
+            self.winner = self.second.name
+            self.first.hp = 0
+
+        if self.second.hp <= 0:
+            self.winner = self.first.name
+            self.second.hp = 0
```

Comment:
There will be a case, when both of them will have <= 0 HP, but in this case the second one would be a winner. Don't think, that's correct.